core/src/main/java/nu/yona/server/goals/service/BudgetGoalDTO.java

```diff
@@ -22,15 +19,15 @@
 	private final int maxDurationMinutes;
 
 	@JsonCreator
-	public BudgetGoalDTO(@JsonProperty("activityCategoryName") String activityCategoryName,
-			@JsonProperty("maxDurationMinutes") int maxDurationMinutes)
+	public BudgetGoalDTO(@JsonProperty(required = true, value = "activityCategoryUrl") String activityCategoryUrl,
+			@JsonProperty(required = true, value = "maxDurationMinutes") int maxDurationMinutes)
 	{
-		this(null, activityCategoryName, maxDurationMinutes, false /* ignored */);
+		this(null, determineActivityCategoryID(activityCategoryUrl), maxDurationMinutes, false /* ignored */);
```

Comment:
Maybe we can move this call to `determineActivityCategoryID` to a `GoalDTO` constructor such that it does not have to be repeated for every goal DTO.